mcu-util/src/messaging/can/isotp.rs

```diff
@@ -256,6 +260,6 @@ impl MessagingInterface for CanIsoTpMessaging {
             _ => return Err(eyre!("Invalid payload")),
         };
 
-        self.send_wait_ack(bytes.as_slice()).await
+        self.send_wait_ack(Arc::new(bytes)).await
```

Comment:
fyi: the [bytes](https://docs.rs/bytes) crate is typically used for reference counted byte buffers, and you'll encounter it throughout much of the tokio ecosystem. There is nothing wrong with using a `Arc<Vec<u8>>` for simple use cases like this though - *if* you need shared references.

However, I don't see why an Arc is needed. Can't you just send the Vec directly through the channel without wrapping it in an Arc?

Reply:
the issue is within `send_wait_ack`, I need to pass the `Vec` to another thread

Reply:
> I need to pass the Vec to another thread
Thats fine: by making `send_wait_ack` take ownership of the vec, you can subsequently send the vec to the other thread. No copy required.